subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -12,11 +12,46 @@
 
 LOG_MODULE_REGISTER(dfu_target_mcuboot, CONFIG_DFU_TARGET_LOG_LEVEL);
 
+#define MAX_FILE_SEARCH_LEN 500
 #define MCUBOOT_HEADER_MAGIC 0x96f3b83d
 
 static struct flash_img_context flash_img;
 static size_t offset;
 
+int dfu_ctx_mcuboot_set_b1_file(char *file, bool s0_active, char **update)
+{
+	if (file == NULL || update == NULL) {
+		return -EINVAL;
+	}
+
+	/* Ensure that 'file' is null-terminated. */
+	if (strnlen(file, MAX_FILE_SEARCH_LEN) == MAX_FILE_SEARCH_LEN) {
+		return -ENOTSUP;
+	}
+
+	/* We have verified that there is a null-terminator, so this is safe */
+	char *space = strstr(file, " ");
+
+	if (space == NULL) {
+		/* Could not find space separator in input */
+		*update = NULL;
```

Comment:
From how this function is used, it seems this line could be changed to `*update = file`;

Reply:
It should not be like that

```
	if (update != NULL) {
		LOG_INF("B1 update, selected file:\n%s", update);
		file = update;
	}
```

If we set it to NULL, the return value in `update` is never used.

Reply:
This is tested by the unit test:
```
	/* update set to null indicates to use original file param */
	dfu_ctx_mcuboot_set_b1_file__update = NULL;
	strcpy(buf, S0_S1);
	err = fota_download_start("something.com", buf);
	zassert_equal(err, 0, NULL);
	zassert_true(strcmp(download_client_start_file, S0_S1) == 0, NULL);

	/* update set to not null indicates to use update for file param */
	dfu_ctx_mcuboot_set_b1_file__update = S1;
	strcpy(buf, S0_S1);
	err = fota_download_start("something.com", buf);
	zassert_equal(err, 0, NULL);
	zassert_true(strcmp(download_client_start_file, S1) == 0, NULL);

```

Reply:
I agree. My point is that returning NULL is the same as returning `file` since that's what it defaults to.

Reply:
I want the caller to be able to see that a space separated `file` has been received. AFAICS, if we go with your solution, there is no way to separate between these cases:
- You got a space separated mcuboot upgrade, and S1 is the active slot
- You got an application upgrade

By looking at the result from calling this function.